src/main/java/roomescape/application/service/ReservationTicketService.java

```diff
@@ -0,0 +1,147 @@
+    private final MemberRepository memberRepository;
+    private final TossPaymentRepository tossPaymentRepository;
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public ReservationTicket saveReservationTicket(ReservationTicketRegisterDto reservationTicketRegisterDto,
+                                                   LoginMember loginMember) {
+        ReservationTicket reservationTicket = createReservationTicket(reservationTicketRegisterDto, loginMember);
+        return reservationTicketRepository.save(reservationTicket);
+    }
+
+    @Transactional(readOnly = true)
+    public void validateReservationTicket
+            (ReservationTicketRegisterDto reservationTicketRegisterDto,
+             LoginMember loginMember) {
+        ReservationTicket reservationTicket = createReservationTicket(reservationTicketRegisterDto,
+                loginMember);
+        assertReservationIsNotDuplicated(reservationTicket);
+    }
+
+    public List<ReservationTicketResponseDto> getReservationTickets() {
+        return reservationTicketRepository.findAll().stream()
+                .map(ReservationTicketResponseDto::new)
+                .toList();
+    }
+
+    public List<MemberReservationResponseDto> getReservationTicketsOfMember(LoginMember loginMember) {
+        List<ReservationTicket> reservationTickets = reservationTicketRepository.findForMember(
+                loginMember.id());
+
+        return reservationTickets.stream()
+                .map(reservationTicket -> new MemberReservationResponseDto(reservationTicket,
+                        tossPaymentRepository.findForReservationTicket(reservationTicket)))
+                .toList();
+    }
+
+    public List<ReservationTicketResponseDto> searchReservationTickets(
+            ReservationSearchDto reservationSearchDto) {
+        Long themeId = reservationSearchDto.themeId();
+        Long memberId = reservationSearchDto.memberId();
+        LocalDate startDate = reservationSearchDto.startDate();
+        LocalDate endDate = reservationSearchDto.endDate();
+
+        return reservationTicketRepository.findForThemeAndMemberInPeriod(
+                        themeId,
+                        memberId,
+                        new Period(startDate, endDate)
+                ).stream()
+                .map(ReservationTicketResponseDto::new)
+                .toList();
+    }
+
+    public void cancelReservationTicket(Long id) {
+        ReservationTicket reservationTicket = reservationTicketRepository.findById(id);
+        reservationTicketRepository.deleteById(id);
+
+        promoteNextWaitingToReservationTicket(reservationTicket);
+    }
+
+    private void promoteNextWaitingToReservationTicket(ReservationTicket reservationTicket) {
+        Optional<Waiting> optionalNextWaiting = waitingRepository.findNextWaiting(
+                reservationTicket.getDate(),
+                reservationTicket.getReservationTime(),
+                reservationTicket.getTheme()
```

Comment:
세 개의 매개변수가 공통적으로 묶여 다닐 것 같은데, 그 의미를 파악하면 Schedule이 될 것 같습니다!
매개변수의 공통된 성질을 Schedule이라는 것으로 묶어 전달하는 것은 어때요? 리팩터링~

```suggestion
                reservationTicket.extractSchedule();
```